### PR TITLE
Splited the install-tools into test/build groups

### PR DESCRIPTION
**Pull request by MeenuyD**
Refactor CI installation process: split install-tools into mockery/testing/build sub-targets, divide install-ci into install-test-ci and install-build-ci, and update workflows accordingly.
#4846 
#4848

### CODE REVIEW

**Comment by slayer321** [comment]
@MeenuyD `DCO` is failing on your PR. Follow the [CONTRIBUTING_GUIDELINES](https://github.com/jaegertracing/jaeger/blob/main/CONTRIBUTING_GUIDELINES.md#creating-a-pull-request). You can read more about DCO [here](https://github.com/jaegertracing/jaeger/blob/main/CONTRIBUTING_GUIDELINES.md#certificate-of-origin---sign-your-work).

File: Makefile [hunk 1]
```diff
@@ -394,14 +394,22 @@ changelog:
 draft-release:
 	./scripts/draft-release.py
 
-.PHONY: install-tools
```
Comment (yurishkuro) [inline Makefile:397]: Please keep this overall target, it's called manually to set up local dev env.

File: Makefile [hunk 2]
```diff
@@ -399,9 +399,27 @@ install-tools:
...
+
+PHONY: install-build-ci
+install-build-ci: install-build-tools
+
+.PHONY: install-test-ci
+install-test-ci: install-test-tools 
 
 .PHONY: install-ci
-install-ci: install-tools
+install-ci: install-tools install-test-ci install-build-ci
```
Comment (yurishkuro) [inline Makefile:421]:
```suggestion
install-ci: install-build-tools
```
and no need to change all workflows, only change the two test workflows to call into `build-test-tools` target

File: Makefile [hunk 3]
```diff
@@ -399,9 +399,27 @@ install-tools:
...
+.PHONY: install-build-tools
+install-build-tools:
+    $(GO) install github.com/josephspurrier/goversioninfo/cmd/goversioninfo@v1.4.0
+    
+
+PHONY: install-build-ci
+install-build-ci: install-build-tools
+
+.PHONY: install-test-ci
+install-test-ci: install-test-tools
```
Comment (yurishkuro) [inline Makefile:418]: this target is not needed

File: Makefile [hunk 4]
```diff
@@ -399,10 +399,28 @@ install-tools:
 	$(GO) install github.com/vektra/mockery/v2@v2.14.0
 	$(GO) install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.52.1
 	$(GO) install mvdan.cc/gofumpt@latest
+	$(GO) install github.com/josephspurrier/goversioninfo/cmd/goversioninfo@v1.4.0
```
[the remark this answers is not in the document]
Comment (yurishkuro) [reply Makefile:402]: remove last three installs and change
```diff
- install-tools:
+ install-tools: install-build-tools install-test-tools
```

File: Makefile [hunk 5]
```diff
@@ -399,9 +399,27 @@ install-tools:
...
+    $(GO) install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.52.1
+    $(GO) install mvdan.cc/gofumpt@latest
+
+.PHONY: install-build-tools
+install-build-tools:
+    $(GO) install github.com/josephspurrier/goversioninfo/cmd/goversioninfo@v1.4.0
+    
+
+PHONY: install-build-ci
+install-build-ci: install-build-tools
```
Comment (yurishkuro) [inline Makefile:415]: this target is not needed

**Comment by MeenuyD** [comment]
Can you please tell me more, please

File: Makefile (hunk 2 above)
Comment (MeenuyD) [reply Makefile:417]: And for the test workflow

**Comment by MeenuyD** [comment]
Hello @yurishkuro sorry for the inconvenience but I am learning can you please tell should i change only these and not change the other workflow files?
```
.PHONY: install-test-tools 
install-test-tools:
    $(GO) install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.52.1

.PHONY: install-build-tools
install-build-tools:
    $(GO) install github.com/josephspurrier/goversioninfo/cmd/goversioninfo@v1.4.0

.PHONY: install-tools
install-tools: install-test-tools install-build-tools


.PHONY: install-ci
install-ci: install-tools

```

**Comment by yurishkuro** [comment]
Install-ci should depend on build and test. Install-tools should also depend on them, and then directly install mockery

**Comment by MeenuyD** [comment]
Is it write
```
.PHONY: install-test-tools 
install-test-tools:
    $(GO) install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.52.1

.PHONY: install-build-tools
install-build-tools:
    $(GO) install github.com/josephspurrier/goversioninfo/cmd/goversioninfo@v1.4.0

.PHONY: install-tools
install-tools: install-test-tools install-build-tools
     $(GO) install github.com/vektra/mockery/v2@v2.14.0

.PHONY: install-ci
install-ci: install-test-tools install-build-tools
```

**Comment by yurishkuro** [comment]
👍 yes

**Comment by MeenuyD** [comment]
Thank you

**Comment by MeenuyD** [comment]
Sorry for all that I have done in this can you please review #4878 once again sorry

**Pull request closed without merging**